src/rust/cryptography-x509-verification/src/policy/mod.rs

```diff
@@ -315,11 +316,18 @@ impl<'a, B: CryptoOps> Policy<'a, B> {
                     Criticality::Agnostic,
                     Some(ee::key_usage),
                 ),
-                // CA/B 7.1.2.7.12 Subscriber Certificate Subject Alternative Name
-                subject_alternative_name: ExtensionValidator::present(
-                    Criticality::Agnostic,
-                    Some(ee::subject_alternative_name),
-                ),
+                subject_alternative_name: if has_subject {
+                    // CA/B 7.1.2.7.12 Subscriber Certificate Subject Alternative Name
+                    ExtensionValidator::present(
+                        Criticality::Agnostic,
+                        Some(ee::subject_alternative_name),
```

Comment:
In addition to checking that subject matches, this function also checks some stuff about criticality and the cert's subject. Do we not need that logic for client certs?

Reply:
Yep, thanks for catching that: CABF has nothing to say about client certs, but 5280 says that the empty subject/SAN criticality rule applies. 

I think the right approach here is going to be to tweak `ee::subject_alternative_name` to pass when the `Policy` doesn't contain a subject, with the expectation that `Policy::server` is always constructed such that a subject is present. Doing that now.